### ISSUE TITLE
Fix `npm start` at root

### ISSUE THREAD

**Issue by mwcz** [spaship/spaship#707]
Running `npm start` at the root of the project used to spin up a working SPAship, but that no longer works.  Investigate
why and restore the previous functionality.  Easy local development setup like this is extremely valuable for new (or returning, like me!)
contributors.

**Comment by mwcz** [comment]
(pasting in some relevant chat logs)

@kunyan 
> the config file of manager is still configurable. but it looks not friendly for local development.
> 
> So we put the config file into the `public` directory, that means user could found it in root directoy (dist). and easy to replace to whatever they want. 
> 
> For openshift user, they could mount a config file within configMap to overwrite this file, even there are any upgrades of manager, they won't need update the config again
> 
> The previous config https://github.com/spaship/spaship/blob/666186671b90f6a2731ac645b009cd663139ff9a/packages/manager/src/EnvParser.ts
we were using the REACT_ENV, which coming from `dotenv`. it looks easy for local development, but if user want to change the config, they have to build source code again. that part should not be user's job, right ?
> 
> If you go to the latest config file, https://github.com/spaship/spaship/blob/master/packages/manager/public/config/manager.js 
> 
> The API config is could be control by end user. the infra user could mount the default config. end user could add custom api hosts whatever they want

@mwcz 
> Thanks @kunyan, that definitely helps me understand why you made the changes.  I can see why rebuilding the source is not optimal for someone deploying a SPAship instance.
> 
> The current approach affects:
> 
> Contributors: negatively
> Deployers: positively
> End-users: unaffected

@mwcz 
> I'm thinking we could merge the two approaches very easily, so Contributors can be positively affected too, without losing any benefit to Deployers.  We could simply update config.js so that it contains the local development defaults, instead of Red Hat's internal URLs.
> 
> That way `npm start` would work, making local development a lot easier, but the file is still easily configurable for end users.

@mwcz 
> And if we wanted, we could take it a step further by moving the config file out of `public` and changing it back to `config.ts`, but updating the app so that it attempts to load a `config.js` file from a standard location (like `/config.js` perhaps).  The `config.ts` could use all the same react env vars as before, but _if_ the Deployer creates a `config.js` file in standard location, it will be automatically loaded and will override anything in `config.ts`.

**Comment by mwcz** [comment]
In addition to the convo above, I just realized that `npm start` at the root probably broke earlier than I thought, because integrating with keycloak necessitated configuration.  In order to resolve this issue, we'd also need a flag to tell the api and manager that they're in a dev environment and shouldn't require communication with a real keycloak server.  I _do_ vaguely remember working on mocking keycloak when I set up the initial integration, but it's very possible I worked in it briefly, then abandoned it for the sake of time savings.
